packages/core/src/dispatcher.ts

```diff
@@ -156,6 +156,27 @@ export class Dispatcher {
     })
   }
 
+  async getIpfsBlock(cid: CID): Promise<Uint8Array> {
+    return await this._shutdownSignal.abortable((signal) => {
+      return this._ipfs.block.get(cid, { signal })
+    })
+  }
+
+  /**
+   * Stores all the blocks in the given CAR file into the local IPFS node.  Uses the blocks API
+   * instead of the dag.import API because the dag.import API requires the CAR file to be a fully
+   * self-contained graph but our CAR file may have references to IPLD objects that aren't included
+   * in the CAR file itself.
+   * @param car
+   */
+  async storeCarFile(car: CAR): Promise<void> {
+    return await this._shutdownSignal.abortable(async (signal) => {
+      for (const block of car.blocks) {
+        await this._ipfs.block.put(block.payload, { signal })
```

Comment:
note to self: pin the CIDs!

Reply:
oh wait, I think the normal stream pinning machinery should take care of pinning these later on.  So I don't think I actually need to worry about pinning here.